app.go

```diff
@@ -526,7 +526,8 @@ func (app *app) runShell(s string, args []string, prefix string) {
 		anyKey()
 	}
 
-	app.ui.loadFile(app.nav, true)
+	// do not call loadFile because of race condition with previewer
+	// app.ui.loadFile(app.nav, true)
```

Comment:
I took this as a puzzle, and here's the problem this creates. 

TLDR: I think the best solution may be to move this line to before line 489 (in the defer that happens for `$` and `!`), unless there's something else that I missed. It may be worth it to put a comment as well about what this is for.

To see the problem, execute `$touch test` and move the cursor to the new `test` file in `lf`. Then, execute `$echo Test >> test`. In the release version of `lf`, the preview will be updated immediately. With your PR, it won't. Same thing will happen if you edit the file with `$vim test`.

I don't think there's much point to doing this refresh if the command is launched with `%` or `|`, though, here, before that command finishes. That also causes the bug you were trying to fix. So, moving this refresh to the codepath for `$` and `!` seems to me like it will solve both problems.

Reply:
I stopped using the previewer in favor of #864 and #865 (with https://github.com/laktak/termplug ... though I need to add lf and Linux to the readme sometimes).

After reading your comments I think this is not the right approach. A better solution might check if the previewed file was updated and only then run the previewer again.

I'm going to close this. Thanks for your help!